utils/cloudinfo/aws.go

```diff
@@ -23,7 +23,10 @@ import (
 )
 
 func onAWS() bool {
-	client := ec2metadata.New(session.New(&aws.Config{}))
+	// the default client behavior retried the operation 4 times with a 5s timeout per attempt.
+	// if you were not on aws, you would block for 20s when invoking this operation.
+	// we reduce retries to 0 to reduce the time this blocks when not on aws.
+	client := ec2metadata.New(session.New(&aws.Config{MaxRetries: aws.Int(0)}))
```

Comment:
I'd hope that the local metadata server can be resolved locally and that 2s
should be good enough.

On Wed, Mar 9, 2016 at 12:25 PM, Derek Carr <notifications@github.com>
wrote:

> In utils/cloudinfo/aws.go
> <https://github.com/google/cadvisor/pull/1152#discussion_r55583061>:
>
> > @@ -23,7 +23,10 @@ import (
> >  )
> >
> >  func onAWS() bool {
> > -	client := ec2metadata.New(session.New(&aws.Config{}))
> > +	// the default client behavior retried the operation 4 times with a 5s timeout per attempt.
> > +	// if you were not on aws, you would block for 20s when invoking this operation.
> > +	// we reduce retries to 0 to reduce the time this blocks when not on aws.
> > +	client := ec2metadata.New(session.New(&aws.Config{MaxRetries: aws.Int(0)}))
>
> @justinsb <https://github.com/justinsb> - do you think its safe to
> decrease the default timeout to 2s?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1152/files#r55583061>.
>